### PR TITLE
First alteration of a text after pasting (ctrl+v) does not trigger 'change' event

### LINKED ISSUE #554: First alteration of a text after pasting (ctrl+v) does not trigger 'change' event
## Are you reporting a feature or a bug?

Bug

## [Check if the issue is already reported](http://docs.ckeditor.com/#!/guide/dev_issues_readme-section-avoid-duplicates)

http://dev.ckeditor.com/ticket/13763

**EDIT:** We have quite a few issue related to change event handling. 
a. I believe 3 below are all duplicates or are very close to each other and one fix could handle all 3 of them
https://dev.ckeditor.com/ticket/13763
https://dev.ckeditor.com/ticket/14675
https://dev.ckeditor.com/ticket/13005
b. When implementing the solution, it would probably be best to create in a way that it allows fixing below issues in the future:
https://dev.ckeditor.com/ticket/13235
https://dev.ckeditor.com/ticket/14329
https://dev.ckeditor.com/ticket/14386
c. There is also a pull request which is panned to be reviewed for CKEditor 4.7.2 - https://github.com/ckeditor/ckeditor-dev/pull/308

## Provide detailed reproduction steps (if any)

1. Paste text into the editor (ctrl +v).
2. Select text.
3. Press delete / backspace.

### Expected result

Change event should be fired as the content of the editor is changed.

### Actual result

Change event is not fired.

## Other details

* Browser: All
* OS: Windows / MacOS
* CKEditor version: 4.5.0 - Repro on (http://ckeditor.com/features)

### PR DESCRIPTION
**Pull request by wojtekw92**
## What is the purpose of this pull request?

Bug fix

## Does your PR contain necessary tests?

All patches which change the editor code must include tests. You can always read more
on [PR testing](http://docs.ckeditor.com/#!/guide/dev_contributing_code-section-tests),
[how to set the testing environment](http://docs.ckeditor.com/#!/guide/dev_tests) and
[how to create tests](http://docs.ckeditor.com/#!/guide/dev_tests-section-creating-your-own-test)
in the official CKEditor documentation.

### This PR contains

- [X] Unit tests
- [X] Manual tests

## What changes did you make?

style, and tags fixes in PR #308 

closes #554

### CODE REVIEW

File: tests/tickets/13763/1.html [hunk 1]
```diff
@@ -0,0 +1,12 @@
+<div id="editor1">
+	<p>Paste something here:</p>
+</div>
+<div id="changes"></div>
+
+<script>
+    var i = 1;
+    var changes = $( "#changes" );
```
Comment (Comandeer) [inline tests/tickets/13763/1.html]: Test is not working as Bender does not include jQuery by default.

File: tests/tickets/13763/1.md [hunk 2]
```diff
@@ -0,0 +1,10 @@
+@bender-tags: 13763, 4.7.1, bug
```
Comment (Comandeer) [inline tests/tickets/13763/1.md]: Please update version tag and add GitHub issue number alongside trac one.

File: plugins/undo/plugin.js [hunk 3]
```diff
@@ -1050,6 +1050,13 @@
 		},
 
 		/**
+		 * Stop ignoring `input` events.
+		 */
+		activateInputEventListener: function() {
+		this.ignoreInputEvent = false;
```
Comment (Comandeer) [inline plugins/undo/plugin.js]: This instruction should be indented.

File: plugins/undo/plugin.js [hunk 4]
```diff
@@ -1080,6 +1087,9 @@
 			editable.attachListener( editable, 'paste', that.ignoreInputEventListener, that, null, 999 );
 			editable.attachListener( editable, 'drop', that.ignoreInputEventListener, that, null, 999 );
 
+			// After paste we need to re-enable input event listener
+			editor.on( 'afterPaste', that.activateInputEventListener, that );
```
Comment (Comandeer) [inline plugins/undo/plugin.js]: I wonder if there shouldn't be same priority for this listener as for ones above.

File: tests/plugins/undo/change.js [hunk 5]
```diff
@@ -180,6 +180,27 @@
 				// After setting text - caret is moved to beginning. We don't care - it does not change nothing.
 				keyTools.keyEvent( keyCodesEnum.LEFT, null, true );
 			} );
+		},
+
+		// 13763
```
Comment (Comandeer) [inline tests/plugins/undo/change.js]: Please add GitHub issue number.

File: tests/plugins/undo/change.js [hunk 6]
```diff
@@ -180,6 +180,27 @@
...
+		'test change event not fired on type after paste': function() {
```
Comment (Comandeer) [inline tests/plugins/undo/change.js]: Isn't the test checking the exact opposite?

File: tests/tickets/13763/1.html (hunk 1 above)
Comment (wojtekw92) [inline tests/tickets/13763/1.html]: fixed

File: tests/tickets/13763/1.md [hunk 7]
```diff
@@ -0,0 +1,10 @@
+@bender-tags: 554, 4.7.1, bug, jquery
```
Comment (wojtekw92) [inline tests/tickets/13763/1.md]: fixed

**Pull request merged**